backend/src/documentation/padding-oracle.md

```diff
@@ -0,0 +1,214 @@
+
+An example for the ciphertext payload is shown in the image below. 
+
+For explanation purposes each block is 4 byte size.
+
+> The payload: 6A211E23**00000000**4529238A
+
+![Padding Oracle Attack Payload](/documentation/img/payload_padding_oracle.png "Padding Oracle Attack Payload")
+
+The value of P1 and P2 is irrelevant because our goal is to decrypt C3 so, we don´t care about their output. P1 has not changed its output since we had only changed the first cipher block. Let's focus into decrypt the C3 block.
+
+With C2 = 00000000, the cipher decrypted to the following plaintext value.
+
+C2 block
+
+> 00 | 00 | 00 | 00
+
+Decrypted Value
+
+> A8 | 09 | F7 | **2C** 
+
+Notice that that the last  byte of C2 decrypts to a plaintext with invalid padding, **2C**.
+
+
+Let's try another value by incrementing the last byte.
+
+C2 block
+
+> 00 | 00 | 00 | **01**
+
+Decrypted value
+
+> A8 | 09 | F7 | **2D** 
+
+
+The output is another padding but with a different decrypted value **2D** since we change the last byte of C2.
+
+If we continue to increment the last byte of C1 (until FF) we will find for sure a value that matches a valid padding sequence. When this value is hit, it will produce a successful response. This value is unique, so, the response will be different than the other 255 values.
+
+Pay attention to the following case.
+
+C2 block
+
+> 00 | 00 | 00 | **2D**
+
+Decrypted value
+
+> A8 | 09 | F7 | **01** 
+
+The last byte of C2 with the value of **2D** decrypted to a valid padding because the last byte of the decrypted value is **01**. According to the PKCS#7 standard, this is valid padding, so, we found the value we want!
+
+> But how can we use this information to decrypt the C3 block?
+
+Now is the easy part. We know the C1 last byte outputs valid padding (**2D**) and we know the plaintext output which is **01**. Now we can infer the value of X, the output of D(E,C3). If you have noticed, we can do that because the X value XORed with C2 outputs the plaintext. At this point, you may know XOR is a commutative operation. 
+
+If,
+
+> Plaintext = C2 XOR X
+
+Then,
+
+> C2 XOR Plaintext = X
+
+And that's how we get the value of X (D(K,C3)).
+
+In our example,
+
+> 0x01 = 0x2D XOR X
+
+So,
+
+> X = 0x2D XOR 0x01
+
+Finally,
+
+> X = 0x2C
+
+0x01 is plaintext[15] and 0x2D is C2[15].
+
+We know the value of X (2C) and we are able to deduce the final value of the plaintext last byte. Simply XOR X with the previous original ciphertext block which is **6A211E23**. Confused? Compare these steps with the diagram of the CBC decryption shown above.
+
+At this time, the last byte of C3 is known; to find the rest of it, we can work backward through the entire block until every byte of X function is cracked, thus letting us decrypt the C3 plaintext one byte at a time. For the other blocks, it is just applying the same method to crack the full message.
+
+## Put the attack in practice
```

Comment:
you skipped a step here.
First you decrypt the ciphertext to know what's the content, and see if you can manipulate.

Also make sure to explain what is the payload received/manipulated, and how you found it in the app